### PR TITLE
Components: Move product icons from My Jetpack into components package

### PR DESCRIPTION
**Pull request by oskosk**
Moves product icons into JS components

#### Changes proposed in this Pull Request:

* Moves product icons into components making them available as imports from `@automattic/jetpack-components`.
* Adds README.md to the icon directory

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Checkout this branch, on a connected site
* Visit My Jetpack
* Confirm the icons are render on the product cards.

### CODE REVIEW

File: projects/js-packages/components/components/product-icons/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	    <BackupIcon />
+        <BoostIcon />
+	    <CheckmarkIcon />
+        <CrmIcon />
+	    <ScanIcon />
+        <SearchIcon />
+	    <StarIcon />
+        <VideoPressIcon />
+    </div>
+)
```
[the remark this answers is not in the document]
Comment (oskosk) [reply projects/js-packages/components/components/product-icons/README.md]: Addressd in 80a7205fe0e6930fb93d1988fce5aaa72421d0ca

**Review by retrofox — COMMENTED** [review]
Moving the icons without any significant changes could make the API ambiguos.
Maybe we should wither rename `getIconBySlug()` by `getProductIconBySlug()`, or rename the icon slug for each icon? For Instead, rename `backup` to `product-backup`.

**Comment by renatoagds** [comment]
I thought to move this and create a unique `Icon` component at `jetpack-components`, which could make us use it in all places:

```
import { Icon } from '@automattic/jetpack-components'
<Icon icon="boost" />
```

I think we can move the current way, and improve next.

**Comment by retrofox** [comment]
> I thought to move this and create a unique `Icon` component at `jetpack-components`, which could make us use it in all places:
> 
> ```
> import { Icon } from '@automattic/jetpack-components'
> <Icon icon="boost" />
> ```
> 
> I think we can move the current way, and improve next.

I did an approach to follow a similar core pattern, but it lead me to think that we may should create a new pkg:

```es6
import { Icon, boost } from '@automattic/jetpack-icons'

function MyCmp() {
  return(
    <Icon icon={ boost } />
  );
}
```

**Comment by oskosk** [comment]
@retrofox how can I reproduce that ambiguity?

**Comment by renatoagds** [comment]
> I did an approach to follow a similar core pattern, but it lead me to think that we may should create a new pkg.

Yep, that's a way to follow. But core already has an `Icon` component that accepts a string: https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/icon#with-a-dashicon

I vote to follow with everything at `jetpack-components`, for me is really confusing to have two packages to go through, every time I got lost 😂

**Comment by oskosk** [comment]
I've addressed some issues. @renatoagds @retrofox could you approve or reject this one? I'd like to be able to use this icons in #23586 without resorting to pure webpack path resolution

**Review by retrofox — APPROVED** [review]

**Comment by oskosk** [comment]
Thanks!

**Comment by oskosk** [comment]
r242549-wpcom

**Comment by oskosk** [comment]
The diff for this PR was problematic. As the PR moved files from My Jetpack into components there was no trace in this diff about the moved files. my-jetpack files are ignored by fusion so the diff didn't care, but it did include the README that was being introduced only in the components version of the dir. cc @Automattic/ground-control

**Comment by anomiex** [comment]
Moved files are one of the annoying edge cases in Fusion. Sounds like you worked it out correctly.

We're working on getting rid of Fusion entirely by committing the built Jetpack extension as a whole to wpcom weekly. See p9dueE-4mF-p2.

**Pull request merged**